src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -968,14 +968,14 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VSMMetadata,
     inserter: Inserter, product: Boolean): VariantSampleMatrix[T] = {
 
     import LocusImplicits.orderedKey
-    def annotate[S](rdd: RDD[(Locus, ((Variant, (Annotation, Iterable[T])), S))],
+    def annotate[S](joinedRDD: RDD[(Locus, ((Variant, (Annotation, Iterable[T])), S))],
       ins: (Annotation, S) => Annotation): OrderedRDD[Locus, Variant, (Annotation, Iterable[T])] = {
-      rdd.mapPartitions({ it =>
+      OrderedRDD(joinedRDD.mapPartitions({ it =>
         it.map { case (l, ((v, (va, gs)), annotation)) => (v, (ins(va, annotation), gs)) }
-      }, preservesPartitioning = true)
-        .asOrderedRDD
+      }, preservesPartitioning = true),
+        rdd.orderedPartitioner)
```

Comment:
> Isn't this code doing literally the same thing?

No.  asOrderedRDD calls cast, which keeps the same partitioner, which in this case is (incorrectly), joinRDD's Locus-keyed parittioner.

apply, on the other hand, doesn't change the data but assumes it is already partitioned according to the partitioner (trust but verify).

 > wouldn't you need to mapMonotonic on the partitioner?

I don't think so.  rdd.partitioner is already a partitioner on Variants.  joinedRDD is partitioned on Locus.  After mapping to recover the variant key, I want to assert that original partitioning is still there.